src/emulator/controller.ts

```diff
@@ -442,20 +442,26 @@ export async function startAll(options: Options, showUI: boolean = true): Promis
     }
 
     const account = getProjectDefaultAccount(options.projectRoot as string | null);
+    // TODO: Go read firebase.json for extensions and add them to emualtableBackends.
+    const emulatableBackends: EmulatableBackend[] = [
+      {
+        functionsDir,
+        env: {
+          ...(options.extensionEnv as Record<string, string> | undefined),
+        },
+        predefinedTriggers: options.extensionTriggers as ParsedTriggerDefinition[] | undefined,
```

Comment:
I agree for the trigger discovery step, but not for actually running the functions. We can't emulate functions with just triggers, we also need to know where the code actually lives so that we can execute it.

Reply:
Ah ha that makes sense. Thanks for explaining.